Makefile

```diff
@@ -0,0 +1,111 @@
+.DEFAULT_GOAL := help
```

Comment:
I don't think we need this large `Makefile`. We can put it on the workflow file for linter and test (and coverage). For source code formatting, let the developers use their tools via the editor's LSP or their preferred way.

Reply:
No one has to use the make file, people can still use what they prefer, this is just to help them making sure their changes will pass the tests before submitting a PR...

But I can remove it, or parts of it if you want.

Reply:
Let's remove it for now and let `golangci-lint` validate the PR request (or they can run it locally). We can always take a look into it later in the future.

In addition to golangci-lint, it would be better to have `.golangci.yml` so we can adjust the linter configuration when needed.

Reply:
Ok